### PR TITLE
Replace many dead links with web.archive.org equivalants.

### PR DESCRIPTION
**Pull request by IoIxD**
There was quite a few dead links, especially at the beginning. This PR replaces them with the most recent web.archive.org snapshot

### CODE REVIEW

**Comment by catwell** [comment]
It would probably be better to replace them with the new links for those projects when they exist, especially when they are lunarmodules projects!

I.e.:

- https://keplerproject.github.io/luadoc/
- https://lunarmodules.github.io/luafilesystem/

The textui docs link (https://nilnor.github.io/textui/docs/) appears to work for me.

Using archive.org for z/OS might make sense, or we may just remove the reference.

**Review by alerque — CHANGES_REQUESTED** [review]
Much like @catwell suggested it would be much better to link to the new homes of these projects rather than to archived versions of their old homes. Even if the exact same page is not available we should adjust the references so that we link to current resources. If anything is really no longer available we should probably just rewrite the docs to avoid the reference and substitute something current or at least make sure it is a minor point.

Thanks for taking the time to contribute. Are you able to update this PR with current links?

**Comment by IoIxD** [comment]
Swear it didn't work for me, but yeah I guess it does. Fixed.

File: manual.md [hunk 1]
```diff
@@ -7,11 +7,11 @@
...
 It is mainly targeted at Lua and documenting Lua APIs, but it can also parse C with according doc comments for documenting Lua modules implemented in C.
 
-It is mostly compatible with [LuaDoc](http://keplerproject.github.io/luadoc/), except that certain workarounds are no longer needed.
+It is mostly compatible with [LuaDoc](https://keplerproject.github.io/luadoc/), except that certain workarounds are no longer needed.
 For instance, it is not so married to the idea that Lua modules should be defined using the `module` function; this is not only a matter of taste since this has been deprecated in Lua 5.2.
 
 Otherwise, the output is very similar, which is no accident since the HTML templates are based directly on LuaDoc.
-You can ship your own customized templates and style sheets with your [own project](https://nilnor.github.io/textui/docs/) (also see Graham Hannington's documentation for [Lua for z/OS](http://lua4z.com/doc/)).
+You can ship your own customized templates and style sheets with your [own project](http://nilnor.github.io/textui/docs/) (also see Graham Hannington's documentation for [Lua for z/OS](https://web.archive.org/web/20191231114636/http://lua4z.com/doc/)).
```
Comment (alerque) [inline manual.md:14]: GitHub Pages supports SSL:

```suggestion
You can ship your own customized templates and style sheets with your [own project](https://nilnor.github.io/textui/docs/) (also see Graham Hannington's documentation for [Lua for z/OS](https://web.archive.org/web/20191231114636/http://lua4z.com/doc/)).
```

**Review by alerque — CHANGES_REQUESTED** [review]

**Review by IoIxD — COMMENTED** [review]
Done.

**Review by alerque — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**